src/main/java/edu/kit/compiler/transform/IRExpressionVisitor.java

```diff
@@ -0,0 +1,222 @@
+                    LogicalOr, NotEqual, LessThan -> {
+                return IRBooleanExpressions.asValue(context, binaryExpressionNode);
+            }
+            case Modulo -> {
+                Node mem = getConstruction().getCurrentMem();
+                Node mod = getConstruction().newMod(mem, lhs, rhs, binding_ircons.op_pin_state.op_pin_state_pinned);
+                Node projRes = getConstruction().newProj(mod, m, Mod.pnRes);
+                Node projMem = getConstruction().newProj(mod, Mode.getM(), Mod.pnM);
+
+                // TODO: map control flow exceptions too?
+                getConstruction().setCurrentMem(projMem);
+                return projRes;
+            }
+            case Addition -> {
+                return getConstruction().newAdd(lhs, rhs);
+            }
+            case Division -> {
+                Node mem = getConstruction().getCurrentMem();
+                Node div = getConstruction().newDiv(mem, lhs, rhs, binding_ircons.op_pin_state.op_pin_state_pinned);
+                Node projRes = getConstruction().newProj(div, m, Div.pnRes);
+                Node projMem = getConstruction().newProj(div, Mode.getM(), Div.pnM);
+
+                // TODO: map control flow exceptions too?
+                getConstruction().setCurrentMem(projMem);
+                return projRes;
+            }
+            case Subtraction -> {
+                return getConstruction().newSub(lhs, rhs);
+            }
+            case Multiplication -> {
+                return getConstruction().newMul(lhs, rhs);
+            }
+            default -> {
+                throw new UnsupportedOperationException("not supported " + binaryExpressionNode.getOperator().toString());
+            }
+        }
+    }
+
+    @Override
+    public Node visit(UnaryExpressionNode unaryExpressionNode) {
+        switch (unaryExpressionNode.getOperator()) {
+            case LogicalNegation -> {
+                return IRBooleanExpressions.asValue(context, unaryExpressionNode);
+            }
+            case ArithmeticNegation -> {
+                ExpressionNode innerNode = unaryExpressionNode.getExpression();
+                Node res = innerNode.accept(this);
+                return getConstruction().newMinus(res);
+            }
+            default -> throw new UnsupportedOperationException("not supported " + unaryExpressionNode.getOperator());
+        }
+    }
+
+    @Override
+    public Node visit(MethodInvocationExpressionNode methodInvocationExpressionNode) {
+        Node mem = getConstruction().getCurrentMem();
+
+        Mode m = context.getTypeMapper().getDataType(methodInvocationExpressionNode.getResultType()).getMode();
+        if (methodInvocationExpressionNode.getObject().isEmpty()) {
+            throw new IllegalArgumentException("object was null on method call");
+        }
+        Node objectAddress = methodInvocationExpressionNode.getObject().get().accept(pointerVisitor);
```

Comment:
You sholdn't call the pointer visitor here, this will crash for most cases. Instead, call `accept(this)` (which might call the pointer visitor recursively)